tests/kernel/ib_tests.pm

```diff
@@ -0,0 +1,207 @@
+    barrier_wait('IBTEST_DONE');
+
+    # just save the dmesg log
+    script_run("dmesg > /tmp/dmesg.txt");
+    upload_logs("/tmp/dmesg.txt");
+
+    # check dmesg for warnings etc.
+    check_dmesg;
+}
+
+sub ibtest_master {
+    my $host1              = get_required_var('IBTEST_IP1');
+    my $host2              = get_required_var('IBTEST_IP2');
+    my $hpc_testing        = get_var('IBTEST_GITTREE', 'https://gitlab.suse.de/NMoreyChaisemartin/hpc-testing.git');
+    my $hpc_testing_branch = get_var('IBTEST_GITBRANCH', 'master');
+
+    # do all test preparations and setup
+    zypper_call('in git twopence bc');
+
+    # create symlinks, the package is (for now) broken
+    assert_script_run('ln -sf /usr/lib64/libtwopence.so.0.3.8 /usr/lib64/libtwopence.so.0');
+    assert_script_run('ln -sf /usr/lib64/libtwopence.so.0.3.8 /usr/lib64/libtwopence.so');
+    # pull in the testsuite
+    assert_script_run("git -c http.sslVerify=false clone $hpc_testing --branch $hpc_testing_branch");
+
+    # create and distribute ssh key
+    assert_script_run('ssh-keygen -b 2048 -t rsa -q -N "" -f ~/.ssh/id_rsa');
+    exec_and_insert_password("ssh-copy-id -o StrictHostKeyChecking=no root\@$host1");
+    script_run("/usr/bin/clear");
+    exec_and_insert_password("ssh-copy-id -o StrictHostKeyChecking=no root\@$host2");
+    script_run("/usr/bin/clear");
+
+    barrier_wait('IBTEST_SETUP');
+    # wait until the two machines under test are ready setting up their local things
+    barrier_wait('IBTEST_BEGIN');
+
+    assert_script_run('cd hpc-testing');
+    assert_script_run("./ib-test.sh $host1 $host2", 1800);
+
+
+    # remove non-printable characters
+    script_run('tr -cd \'\11\12\15\40-\176\' < results/TEST-ib-test.xml > /tmp/results.xml');
+    parse_extra_log('XUnit', '/tmp/results.xml');
+
+    script_run("scp -o StrictHostKeyChecking=no root\@$host1:/tmp/dmesg.txt /tmp/dmesg_slave1.txt");
+    script_run("scp -o StrictHostKeyChecking=no root\@$host2:/tmp/dmesg.txt /tmp/dmesg_slave2.txt");
+
+    check_dmesg('/tmp/dmesg_slave1.txt');
+    check_dmesg('/tmp/dmesg_slave2.txt');
+
+    barrier_wait('IBTEST_DONE');
+    barrier_destroy('IBTEST_SETUP');
+    barrier_destroy('IBTEST_BEGIN');
+    barrier_destroy('IBTEST_DONE');
+}
+
+sub run {
+    my $role = get_required_var('IBTEST_ROLE');
+
+    select_console 'root-ssh';
+    zypper_call('ar -f -G ' . get_required_var('GA_REPO'));
+
+    if ($role eq "IBTEST_MASTER") {
+        zypper_call('ar -f -G ' . get_required_var('DEVEL_TOOLS_REPO'));
+    }
+
+    zypper_call('--gpg-auto-import-keys ref');
+
+    if ($role eq "IBTEST_MASTER") {
+        ibtest_master;
+    }
+    elsif ($role eq 'IBTEST_SLAVE1' || $role eq 'IBTEST_SLAVE2') {
+        ibtest_slave;
+    }
+
+    power_action('poweroff', observe => 0, keepconsole => 0);
+}
+
+sub test_flags {
+    return {fatal => 1};
+}
+
+sub post_fail_hook {
+    script_run('tr -cd \'\11\12\15\40-\176\' < results/TEST-ib-test.xml > /tmp/results.xml');
```

Comment:
A short explanation of this magic set of characters and where they come from would be nice

Reply:
Sure, I could add a comment. This line just removes non-printable characters.